### PR TITLE
Add ServerConnectEvent Reason API

### PR DESCRIPTION
**Pull request by Mystiflow**

### CODE REVIEW

**Comment by md-5** [comment]
Looks fine to me

File: api/src/main/java/net/md_5/bungee/api/event/ServerConnectEvent.java [hunk 1]
```diff
@@ -61,10 +61,20 @@ public ServerConnectEvent(ProxiedPlayer player, ServerInfo target, Reason reason
...
+        SERVER_DOWN_REDIRECT,
         /**
          * Redirecting to another server when kicked from original server.
          */
         KICK_REDIRECT,
+        /**
+         * Redirecting to another server when kicked from server during the
+         * handshake.
+         */
+        KICK_REDIRECT_HANDSHAKE,
```
Comment (md-5) [inline api/src/main/java/net/md_5/bungee/api/event/ServerConnectEvent.java]: Why does this need to be distinguished?

Comment (Mystiflow) [reply api/src/main/java/net/md_5/bungee/api/event/ServerConnectEvent.java]: Usually kicks during the login are bans, whitelists, etc. Just allows for more granular control really.

Comment (md-5) [reply api/src/main/java/net/md_5/bungee/api/event/ServerConnectEvent.java]: Yeah but it also complicates the API usage (eg in switches) for granular control that I'm not sure is required

Comment (Mystiflow) [reply api/src/main/java/net/md_5/bungee/api/event/ServerConnectEvent.java]: Okay, I reverted the change.

**Pull request closed without merging**